openshift/default-catalog-consistency/test/utils/utils.go

```diff
@@ -0,0 +1,73 @@
+package utils
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"sigs.k8s.io/yaml"
+	"strings"
+
+	apiv1 "github.com/operator-framework/operator-controller/api/v1"
+)
+
+// LoadImagesFromCatalogs reads the catalogs from the files used and returns a list of image references.
+func LoadImagesFromCatalogs() ([]string, error) {
+	var images []string
+	catalogsPath := "../../../catalogd/kustomize/overlays/openshift/catalogs"
+
+	// Check if the directory exists first
+	if _, err := os.Stat(catalogsPath); os.IsNotExist(err) {
+		return nil, fmt.Errorf("catalogs path %s does not exist", catalogsPath)
+	}
+
+	err := filepath.WalkDir(catalogsPath, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if d.IsDir() || (!strings.HasSuffix(d.Name(), ".yaml")) {
+			return nil
+		}
+
+		content, err := os.ReadFile(path)
+		if err != nil {
+			return fmt.Errorf("read %s: %w", path, err)
+		}
+
```

Comment:
maybe we could ignore and log any content that doesn't contain, e.g. `Kind: ClusterCatalog` that way if any other yamls are commingled in the directory we don't error our unnecessarily?

Reply:
Good call 👍